RecoEgamma/EgammaElectronAlgos/src/PixelHitMatcher.cc

```diff
@@ -173,7 +217,7 @@ std::vector<SeedWithInfo> PixelHitMatcher::operator()(const std::vector<const Tr
           auto idx2 = std::distance(hits.first, it2);
           const DetId id2 = it2->geographicalId();
           const GeomDet *geomdet2 = it2->det();
-          const std::pair<const GeomDet *, GlobalPoint> det_key(geomdet2, hit1Pos);
+          const auto det_key = std::make_pair(geomdet2->gdetIndex(), hit1Pos);
```

Comment:
the logic of the map filling is different now.
Is the gdetIndex unique?

Reply:
I honestly didn't really check, I just did this for the sake of unifying with the HLT code which does that. I trust Sams modification there. It's certainly better than C-casting the GeomDet [1] to a `long long` and hoping for the best!

[1] https://github.com/cms-sw/cmssw/blob/master/Geometry/CommonDetUnit/interface/GeomDet.h#L102